completed/camunda-8.js

```diff
@@ -0,0 +1,205 @@
+
+  // These settings come from your .env file.
+  const camundaApiUrl = process.env.ZEEBE_BASE_URL;
+
+  // This is the API endpoint to add a new client to a cluster.
+  const url = `${camundaApiUrl}/groups`;
+
+  // Configure the API call.
+  const options = {
+    method: "POST",
+    url,
+    headers: {
+      Accept: "application/json",
+      Authorization: `Bearer ${accessToken}`
+    },
+    data: {
+      // The body contains information about the new group.
+      groupName : groupName
+    }
+  };
+
+  try {
+    const response = await axios(options);
+
+    // Process the results from the API call.
+    const newGroup = response.data;
+
+    // Emit new group to output.
+    console.log(
+      `Group added! Name: ${newGroup.name}. Key: ${newGroup.groupKey}.`
+    );
+  } catch (error) {
+    // Emit an error from the server.
+    console.error(error.message);
+  }
+}
+
+// An action that assigns a user to a group by a key.
+async function assignUser([groupKey, userKey]) {
+  // Every request needs an access token.
+  const accessToken = await getAccessToken(authorizationConfiguration);
+
+  // These settings come from your .env file.
+  const camundaApiUrl = process.env.ZEEBE_BASE_URL;
+
+  // This is the API endpoint to assign a user to a group.
+  const url = `${camundaApiUrl}/groups/${groupKey}/users/${userKey}`;
+
+  // Configure the API call.
+  const options = {
+    method: "POST",
+    url,
+    headers: {
+      Accept: "application/json",
+      Authorization: `Bearer ${accessToken}`
+    },
+  };
+
+  try {
+    // Call the add endpoint.
+    const response = await axios(options);
+
+    // Process the results from the API call.
+  if (response.status === 204) {
+    console.log(`Group assigned to ${userKey}.`);
+  } else {
+    // Emit an unexpected error message.
+    console.error("Unable to assign this user!");
+  }
+  } catch (error) {
+    // Emit an error from the server.
+    console.error(error.message);
+  }
+}
+
+// An action that retrieves assigned member keys within a group.
+async function retrieveGroup([groupKey]) {
+  // Every request needs an access token.
+  const accessToken = await getAccessToken(authorizationConfiguration);
+
+  // These settings come from your .env file.
+  const camundaApiUrl = process.env.ZEEBE_BASE_URL;
+
+  // This is the API endpoint to list all assigned member keys within a group.
+  const url = `${camundaApiUrl}/groups/${groupKey}`;
+
+  // Configure the API call.
+  const options = {
+    method: "GET",
+    url,
+    headers: {
+      Accept: "application/json",
+      Authorization: `Bearer ${accessToken}`
+    }
+  };
+
+  try {
+    // Call the endpoint.
+    const response = await axios(options);
+
+    // Process the results from the API call.
+    const results = response.data;
+
+    // Emit clients to output.
+    results.forEach(x => console.log(`Name: ${x.name}; ID: ${x.assignedMemberKeys
+    }`));
+// Not sure how to format the above -- is forEach needed??
```

Comment:
One lingering comment here I wasn't sure how to structure.

Reply:
We chatted in Slack and I'm going to push up a commit that changes this PR significantly, but to answer this question first -- 

No, you wouldn't need `forEach`. If you look at [the docs for this endpoint](https://docs.camunda.io/docs/next/apis-tools/camunda-api-rest/specifications/get-group/#responses), you can see the shape of the response is a single object, not an array. So you would have wanted something like this here: 

```js
console.log(`Name: ${results.name}; Key: ${results.groupKey}`)
```